crates/starknet-devnet-server/src/api/http/endpoints/time.rs

```diff
@@ -1,30 +1,42 @@
 use axum::{Extension, Json};
+use starknet_core::error::Error;
 
 use crate::api::http::error::HttpApiError;
-use crate::api::http::models::{IncreaseTimeResponse, SetTimeResponse, Time};
+use crate::api::http::models::{IncreaseTime, IncreaseTimeResponse, SetTime, SetTimeResponse};
 use crate::api::http::{HttpApiHandler, HttpApiResult};
 
 pub async fn set_time(
-    Json(data): Json<Time>,
+    Json(data): Json<SetTime>,
     Extension(state): Extension<HttpApiHandler>,
 ) -> HttpApiResult<Json<SetTimeResponse>> {
     let mut starknet = state.api.starknet.write().await;
+    let generate_block = data.generate_block.unwrap_or(true);
+
     starknet
-        .set_time(data.time)
+        .set_time(data.time, generate_block)
         .map_err(|err| HttpApiError::BlockSetTimeError { msg: err.to_string() })?;
 
     let last_block = starknet.get_latest_block();
     match last_block {
         Ok(block) => Ok(Json(SetTimeResponse {
             block_timestamp: block.timestamp().0,
-            block_hash: block.block_hash(),
+            block_hash: Some(block.block_hash()),
         })),
+        Err(Error::NoBlock) => {
```

Comment:
Why would `NoBlock` ever be thrown?

Reply:
- start devnet
- set_time with no block
- get latest block

Reply:
I also added this comment that this section can be removed later:

            // Handle case when generate_block is false and there is no latest block, this case can
            // be removed once the genesis block is implemented

Reply:
fixed in 56a5f95ac63ceaf41493898b9f87861e7545d79e